### DISCUSSION TITLE
Feature Request: Mailbox encryption / dovecot MailCrypt

### DISCUSSION THREAD

**Discussion by besendorf** [modoboa/modoboa#2070] [Ideas]
It would be a huge privacy gain for modoboa if it supported user mailbox encryption. Dovecot offers a plugin called [MailCrypt](https://doc.dovecot.org/configuration_manual/mail_crypt_plugin/#mail-crypt-plugin) which provides this functionality. It offers two modes. Global keys and folder keys. 
Global keys encrypt all mail accounts with the same key, this is useful when you want to save the mailboxes on a different server or Storage provider (Amazon S3, etc.). 
Folder keys uses a different key for each folder (user mailbox) derived from the users IMAP password. This means administrators are unable to read users mails, which is a huge privacy gain but this is also a security gain, as an attacker who gained access to the mailboxes also cant decrypt the emails(unless he controls the server long enough that he intercepts IMAP passwords).
There is also a paper explaining mailbox encryption in detail and analyses the implications for performance. Unfortunatly its in German:
https://gitlab.com/bifi/mailboxencryption_thesis/-/blob/master/thesis.pdf

**Comment by jeromelebleu** [comment]
I would actually really like to find some time to work on it - if no one else does it. Does anybody want to join? I just don't know yet how it is working and I don't speak German unfortunately. But I planned to see how it is done in userli. Any resources, documentations and help is welcome so!
